### PR TITLE
Add "darwin" to the list of valid macOS architectures

### PR DESCRIPTION
**Pull request by keith**
This is an alternative solution to https://github.com/bazelbuild/bazel/pull/7062 that is more backwards compatible

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/rules/apple/ApplePlatform.java [hunk 1]
```diff
@@ -53,7 +53,7 @@
   private static final ImmutableSet<String> TVOS_DEVICE_TARGET_CPUS =
       ImmutableSet.of("tvos_arm64");
   private static final ImmutableSet<String> MACOS_TARGET_CPUS =
-      ImmutableSet.of("darwin_x86_64");
+      ImmutableSet.of("darwin_x86_64", "darwin");
```
Comment (sergiocampama) [inline src/main/java/com/google/devtools/build/lib/rules/apple/ApplePlatform.java]: can you add a note here on why `darwin` is present? Perhaps linking to the other PR? Just so it's easier to follow what happened when someone takes the cleanup task of this.

Comment (keith) [reply src/main/java/com/google/devtools/build/lib/rules/apple/ApplePlatform.java]: Done

**Comment by keith** [comment]
Everything is green, looks like we should be good to go here! @katre can you help me land this?

**Comment by keith** [comment]
@katre friendly ping!

**Comment by katre** [comment]
Yes, I can handle importing this. My apologies for not catching the first time.

**Comment by keith** [comment]
Thanks! No worries!

**Comment by keith** [comment]
Thanks!

**Pull request closed without merging**